test/client/jest.config.js

```diff
@@ -10,10 +10,10 @@ module.exports = {
 	},
 	modulePaths: [ '<rootDir>/test/', '<rootDir>/client/', '<rootDir>/client/extensions/' ],
```

Comment:
might need to add packages here too?

Reply:
Something isn't working correctly, the [package tests are failing](https://circleci.com/gh/Automattic/wp-calypso/131410#tests/containers/5) and not generating test report on CircleCI 😕 

I'll push a fix for the missing import:

```
 FAIL  packages/tree-select/test/index.js
  ● Test suite failed to run

    Cannot find module '../' from 'index.js'

       5 |  * Internal dependencies
       6 |  */
    >  7 | import treeSelect from '../';
         |                                ^
       8 | 
       9 | describe( 'index', () => {
      10 | 	describe( '#treeSelect', () => {

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:221:17)
      at Object.<anonymous> (packages/tree-select/test/index.js:7:32)
```

---

Might be good info here: https://github.com/facebook/jest/issues/3112

In particular, [this looks interesting](https://github.com/facebook/jest/issues/3112#issuecomment-398581705). Each package (Jest project) provides its own config:

```js
const base = require("./jest.config.base.js");

module.exports = {
    ...base,
    projects:
    [
        "<rootDir>/packages/*/jest.config.js"
    ],
    coverageDirectory: "<rootDir>/coverage/"
};
```

Ours might look like:

```
{
    projects:
    [
        "<rootDir>/test/client/jest.config.js",
        "<rootDir>/test/server/jest.config.js",
        "<rootDir>/packages/*/jest.config.js"
    ],
}
```

Should client and package tests be mixed together? That's the current approach in this PR.

This seems to be working, so we can proceed and iterate with smaller, targeted PRs to improve the testing setup.

Reply:
Ah, fun. The tests failed because `lerna bootstrap` hadn't been run, hence the build hadn't been run for tree-select. It was working for me locally because they had and the test was resolving `../` using the `main` in the `package.json` for tree-select (which points at dist/...)...

So one question is why bootstrap hadn't been run up in circle. 

I think the questions on how best to run the tests are totally valid. I'd love to see it separated out from the calypso client tests.

Reply:
> So one question is why bootstrap hadn't been run up in circle.

The `npm` cache on Circle matches checksums of `.nvmrc` and `npm-shrinkwrap.json`. If there is a cached version where those haven't changed, _`npm ci` is completely skipped_:

https://github.com/Automattic/wp-calypso/blob/58564744a96da9a8599fc7eed2e8da3e7d4fd09c/.circleci/config.yml#L110-L118

I believe that means the `postinstall` that's been added here which runs `learna bootstrap` is never run on CI. Do we need to ensure `npm ci` runs, or can we just ensure `lerna bootstrap` does run?

```bash
      if [ ! -d node_modules ]; then
        npm ci
      else
        lerna bootstrap
      fi
```